File: observer_java/src/main/java/com/yb/observer_java/listener/SubjectListener.java
package com.yb.observer_java.listener;

/**
 * 类说明：被观察者接口
 *
 * @author 裕博
 */
public interface SubjectListener {
    /**
     * 添加
     */
    void add(ObserverListener observerListener);

    /**
     * 通知的内容
     */
    void notifyObserver(String count);

    /**
     * 删除
     */
    void remove(ObserverListener observerListener);
}

File: observer_kotlin/src/main/java/com/yb/observer_kotlin/listener/SubjectListener.kt
package com.yb.observer_kotlin.listener

/**
 * 类说明：被观察者接口
 *
 * @author 裕博
 */
interface SubjectListener {
    /**
     * 添加
     */
    fun add(observerListener: ObserverListener)

    /**
     * 通知的内容
     */
    fun notifyObserver(count: String)

    /**
     * 删除
     */
    fun remove(observerListener: ObserverListener)
}
